test.js

```diff
@@ -0,0 +1,168 @@
+  number: 478,
+  title: 'Full text search with mongoDB',
+  user: {
+    login: 'Dubask',
+    id: 30698018,
+    node_id: 'MDQ6VXNlcjMwNjk4MDE4',
+    avatar_url: 'https://avatars.githubusercontent.com/u/30698018?v=4',
+    gravatar_id: '',
+    url: 'https://api.github.com/users/Dubask',
+    html_url: 'https://github.com/Dubask',
+    followers_url: 'https://api.github.com/users/Dubask/followers',
+    following_url: 'https://api.github.com/users/Dubask/following{/other_user}',
+    gists_url: 'https://api.github.com/users/Dubask/gists{/gist_id}',
+    starred_url: 'https://api.github.com/users/Dubask/starred{/owner}{/repo}',
+    subscriptions_url: 'https://api.github.com/users/Dubask/subscriptions',
+    organizations_url: 'https://api.github.com/users/Dubask/orgs',
+    repos_url: 'https://api.github.com/users/Dubask/repos',
+    events_url: 'https://api.github.com/users/Dubask/events{/privacy}',
+    received_events_url: 'https://api.github.com/users/Dubask/received_events',
+    type: 'User',
+    site_admin: false
+  },
+  labels: [{ name: 'testLabel' }],
+  state: 'open',
+  locked: false,
+  assignee: null,
+  assignees: [],
+  milestone: null,
+  comments: 3,
+  created_at: '2021-06-30T09:12:31Z',
+  updated_at: '2021-06-30T13:17:56Z',
+  closed_at: null,
+  author_association: 'NONE',
+  active_lock_reason: null,
+  body:
+    'Im trying to get a Full Text Search working through Fastify and Mongoose. But it returns an empty array.\r\n' +
+    '\r\n' +
+    'There is no info online about this, so I just played with it for a while. in db.ts I set the:\r\n',
+  reactions: {
+    url: 'https://api.github.com/repos/fastify/help/issues/478/reactions',
+    total_count: 0,
+    '+1': 0,
+    '-1': 0,
+    laugh: 0,
+    hooray: 0,
+    confused: 0,
+    heart: 0,
+    rocket: 0,
+    eyes: 0
+  },
+  timeline_url: 'https://api.github.com/repos/fastify/help/issues/478/timeline',
+  performed_via_github_app: null,
+  score: 1
+}
+
+const mockIssue1 = {
+  ...mockIssue,
+  repository_url: 'https://api.github.com/repos/fastify/test1',
+  html_url: 'https://github.com/fastify/help/issues/478'
+}
+
+// app was falling over because of incorrectly parsing this url previously
+const mockIssue2 = {
+  ...mockIssue,
+  html_url: 'https://github.com/fastify/help/issues/4781',
+  repository_url: 'https://api.github.com/repos/fastify/.test2'
+}
+
+const mockIssuesData = {
+  total_count: 40,
+  incomplete_results: false
+}
+
+const mockSearchIssuesAndPullRequests = {
+  status: 200
+}
+
+tap.test('tests the "/api/find-issues" route', async t => {
+  const searchIssuesStub = sinon.stub()
+
+  searchIssuesStub.onCall(0).returns({
+    ...mockSearchIssuesAndPullRequests,
+    data: { ...mockIssuesData, items: [mockIssue1] }
+  })
+  searchIssuesStub.onCall(1).returns({
+    ...mockSearchIssuesAndPullRequests,
+    data: { ...mockIssuesData, items: [mockIssue2] }
+  })
+
+  const build = t.mock('./app', {
+    './fetchIssues': {
+      fetchIssues,
```

Comment:
Not too happy with how I have done this - tap seems to be mocking the whole module and so when I removed this line, it couldn't find the `fetchIssues` function. Not how I understood this should work. Open to suggestions on how to do this in a cleaner way